packages/chat-widget/src/index.tsx

```diff
@@ -352,6 +357,75 @@ const isInputDisabled = (responses: Response[]): boolean => {
   return new URLSearchParams(payload).get("nlx:input-disabled") === "true";
 };
 
+const findActiveUpload = (responses: Response[]): UploadUrl | null => {
+  const lastBotResponse = findLast(
+    (response): response is BotResponse => response.type === "bot",
+    responses,
+  );
+  if (lastBotResponse == null) {
+    return null;
+  }
+  return lastBotResponse.payload.metadata?.uploadUrls?.[0] ?? null;
+};
+
+interface ImageUploadProps {
+  upload: UploadUrl;
+}
+
+type UploadStatus = "empty" | "uploading" | "uploaded";
+
+const ImageUpload: FC<ImageUploadProps> = (props) => {
+  const [status, setStatus] = useState<UploadStatus>("empty");
+
+  if (status === "uploading") {
+    return <C.PendingMessageDots />;
+  }
+
+  if (status === "uploaded") {
+    return (
+      <C.UploadSuccess>
+        <CheckIcon />
+      </C.UploadSuccess>
+    );
+  }
+
+  const handleChange = async (
+    ev: ChangeEvent<HTMLInputElement>,
+  ): Promise<void> => {
+    const file = ev.target.files?.[0];
+    if (file == null) {
+      return;
+    }
+    setStatus("uploading");
+    try {
+      await fetch(props.upload.url, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "image/jpeg",
+        },
+        body: new Blob([file], { type: file.type }),
```

Comment:
I think this is unnecessary. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/File):

> A File object is a specific kind of [Blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob), and can be used in any context that a Blob can. In particular, the following APIs accept both Blobs and File objects:
> 
> ...
> - the [body](https://developer.mozilla.org/en-US/docs/Web/API/RequestInit#body) option to [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/Window/fetch)


So I think this can just be:

```suggestion
        body: file,
```